### ISSUE TITLE
browser phone with wss

### ISSUE THREAD

**Issue by edimondi** [InnovateAsterisk/Browser-Phone#168]
hi, i am trying to use browser phone with wss, or tls if its better but for some reason cant make it work. Because i am using now with ws and i am not sure if the connection is secure with ws, thanks

**Comment by edimondi** [comment]
in browser phone the websocket path i have /ws and is not working with /wss but i am using browser phone with https so i dont know why /ws is working over https

**Comment by InnovateAsterisk** [comment]
The path does not determine the protocol. It only refers to the location web socket service on the server. With Asterisk, its only (hard coded) at /ws

If you type ```> http show status``` in Asterisk it will show you:

```
HTTP Server Status:
Prefix: 
Server: Asterisk/certified/16.8-cert3
Server Enabled and Bound to 0.0.0.0:8080

Enabled URI's:
/httpstatus => Asterisk HTTP General Status
/ws => Asterisk HTTP WebSocket

Enabled Redirects:
  None.
```

as you can see above the path to websocket is /ws

In my example above, you can also see that (in this case) my server is hosting this location as http on port 8080, and that I don't have TLS enabled... but that's because I have this Asterisk box behind a load balancer, and the Load Balancer is terminating my TLS connections, and proxying the non-tls connection into asterisk. I have recently done a video on how to do that: 

https://www.innovateasterisk.com/s2e1-webrtc-reverse-proxy/

(For simplicity, and testing, this proxy step isn't necessary)

**Comment by edimondi** [comment]
in the browser phone console network tab i see this: **Request URL: wss://mydomain:8089/ws** , is this saying that i am using /wss ? Thanks

**Comment by InnovateAsterisk** [comment]
The request url, is much like a regular web site url, in that ```wss://mydomain:8089/ws```
is made up from: the **protocol** ```wss://```, the **address** ```mydomain``` with its **port** (if omitted will use the default for the protocol) and then finally the server **path** ```/ws``` that would specify the exact resource you are looking for.

As an interesting quirk, you can actually test your web socket url in your browser address bar, however if you try:
```
wss://mydomain:8089/ws
```
in Chrome address bar, Chrome will give you a protocol error, but if you change it to:
```
https://mydomain:8089/ws
```
Chrome will attempt the connection, but of course the server will fail to connect since it will be expecting to be upgraded to websocket.

**Comment by InnovateAsterisk** [comment]
With Developer Tools open, tab over to security, all the certificate details are there including validation of the certificate.

**Comment by edimondi** [comment]
yes there is a certificate but is the connection secure between browser phone and asterisk
